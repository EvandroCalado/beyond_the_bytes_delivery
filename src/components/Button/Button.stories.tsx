import { Meta, StoryFn } from '@storybook/react';

import { Button, ButtonProps } from '.';

export default {
  title: 'Components/Button',
  component: Button,
} as Meta;

export const Buttons: StoryFn<ButtonProps> = () => (
  <div style={{ display: 'flex', gap: '1rem', flexWrap: 'wrap' }}>
    <Button>primary</Button>

    <Button variant="secondary">secondary</Button>

    <Button disabled>disabled</Button>
  </div>
);
