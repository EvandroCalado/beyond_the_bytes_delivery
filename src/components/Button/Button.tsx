import { ButtonHTMLAttributes, FC } from 'react';

import * as S from './Button.styles';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  variant?: 'primary' | 'secondary';
}

export const Button: FC<ButtonProps> = ({ children, variant, ...rest }) => {
  return (
    <S.Button variant={variant} {...rest}>
      {children}
    </S.Button>
  );
};
