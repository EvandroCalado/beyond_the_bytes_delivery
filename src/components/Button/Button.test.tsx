import { render, screen } from 'utils';

import theme from 'styles/theme';

import { Button } from './Button';

describe('Button', () => {
  it('should render default', () => {
    render(<Button>button</Button>);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('button');
    expect(button).toHaveStyle({
      backgroundColor: theme.colors.primary,
      color: theme.colors.white,
    });
  });

  it('should render with variant secondary', () => {
    render(<Button variant="secondary">button</Button>);

    const button = screen.getByRole('button');

    expect(button).toHaveStyle({
      backgroundColor: theme.colors.white,
      color: theme.colors.primary,
    });
  });

  it('should render a snapshot', () => {
    render(<Button variant="secondary">button</Button>);

    const button = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });
});
