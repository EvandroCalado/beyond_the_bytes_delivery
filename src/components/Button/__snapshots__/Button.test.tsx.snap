// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  gap: .8rem;
  padding: 1.6rem;
  border: 1px solid #131118;
  border-radius: .8rem;
  font-weight: 600;
  text-transform: capitalize;
  outline: none;
  cursor: pointer;
  transition: all 150ms ease-in-out;
  background-color: #ffffff;
  color: #131118;
}

.c0:hover {
  filter: brightness(0.95);
}

.c0:active {
  transform: scale(1.03);
}

.c0:disabled {
  opacity: 0.1;
  pointer-events: none;
}

<button
  class="c0"
  variant="secondary"
>
  button
</button>
`;
